### PR TITLE
Update Dockerfiles

### PR DESCRIPTION
**Pull request by navidshaikh**

### CODE REVIEW

**Comment by navidshaikh** [comment]
/retest

setup issues

**Comment by navidshaikh** [comment]
The changeset brings:
 - Building kn-cli-artifacts image with signed binaries from lookaside cache
 - New base image
 - Updated labels
 - Improved layering in images
 - New Python script for serving kn cli artifacts 
 - kn-cli-artifacts image entry point now set to Python script serving artifacts
 - Updated kn version tag to `0.13.2`

**Comment by navidshaikh** [comment]
/hold

PR against master to be merged  https://github.com/openshift/knative-client/pull/329

File: container.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+go:
+  modules:
+  - module: github.com/knative/client
```
Comment (warrenvw) [inline container.yaml:4]: We've been using the midstream repo in `container.yaml`. Not the upstream repo.

**Comment by navidshaikh** [comment]
The PR keep failing CI after the nightly CI push to release-next and causing merge conflicts.

**Pull request closed without merging**